signing.go

```diff
@@ -254,11 +255,26 @@ func (ctx *genericSigner) Sign(payload []byte) (*JSONWebSignature, error) {
 
 		serializedProtected := mustSerializeJSON(protected)
 
-		input := []byte(fmt.Sprintf("%s.%s",
-			base64.RawURLEncoding.EncodeToString(serializedProtected),
-			base64.RawURLEncoding.EncodeToString(payload)))
+		var input bytes.Buffer
 
-		signatureInfo, err := recipient.signer.signPayload(input, recipient.sigAlg)
+		input.WriteString(base64.RawURLEncoding.EncodeToString(serializedProtected))
+		input.WriteByte('.')
+
+		enc := true
```

Comment:
Same comment as above, let's call this something other than "enc". Could be confused to mean "encrypt" for example.